fili-core/src/main/java/com/yahoo/bard/webservice/data/config/metric/makers/MetricMaker.java

```diff
@@ -184,56 +187,108 @@ protected TemplateDruidQuery getMergedQuery(List<String> names) {
+     *     <dt>If it's already a numeric post aggregator
+     *     <dd>simply return it
+     * </dl>
+     *
+     * @param field  The field being coerced
+     *
+     * @return A post aggregator representing a number field value
+     */
+    protected static PostAggregation getNumericField(MetricField field) {
+        // If the field is a sketch, wrap it in a sketch estimate, if it's an aggregation, create a post aggregation
+        // Otherwise it is a number post aggregation already
+        return field.isSketch() ?
+                FieldConverterSupplier.sketchConverter.asSketchEstimate(field) :
+                field instanceof Aggregation ?
+                        new FieldAccessorPostAggregation((Aggregation) field) :
+                        (PostAggregation) field;
     }
 
     /**
-     * Get the sketch post aggregation for a sketch field.
+     * Prepare a post aggregation for a field expecting a sketch value.
+     * <p>
+     * The post-agg is created per the following heuristics:
+     * <dl>
+     *     <dt>If it's an aggregator
+     *     <dd>wrap it in a field accessor
+     *     <dt>If it's a sketch estimate
+     *     <dd>unwrap the sketch estimate
+     *     <dt>If it's already a sketch post aggregator
+     *     <dd>simply return it
+     *     <dt>Otherwise</dt>
+     *     <dd>This is an illegal field</dd>
+     * </dl>
+     *
+     * @param fieldName  The name for the aggregation or post aggregation column being gotten
      *
-     * @param fieldName  The name of an aggregation or post aggregation
+     * @return A post aggregator representing a number field value
      *
-     * @return A post aggregation referencing a sketch value
+     * @deprecated use the static version {@link #getSketchField(MetricField)} by preference
      */
+    @Deprecated
     protected PostAggregation getSketchField(String fieldName) {
         // Get the field
-        MetricField field = metrics.get(fieldName).getMetricField();
+        return getSketchField(metrics.get(fieldName).getMetricField());
+    }
 
+    /**
+     * Prepare a post aggregation for a field expecting a sketch value.
+     * <p>
+     * The post-agg is created per the following heuristics:
+     * <dl>
+     *     <dt>If it's an aggregator
+     *     <dd>wrap it in a field accessor
+     *     <dt>If it's a sketch estimate
+     *     <dd>unwrap the sketch estimate
+     *     <dt>If it's already a sketch post aggregator
+     *     <dd>simply return it
+     *     <dt>Otherwise</dt>
+     *     <dd>This is an illegal field</dd>
+     * </dl>
+     *
+     * @param field  The field being coerced
+     *
+     * @return A post aggregator representing a number field value
+     */
+    protected static PostAggregation getSketchField(MetricField field) {
         // SketchEstimatePostAggregations are just wrappers for the actual PostAggregation we want to return
         if (field instanceof SketchEstimatePostAggregation) {
```

Comment:
It seems like `SketchEstimatePostAggregation` is deprecated, should we log something here to take note of it?